### PR TITLE
Refueling fuel containers

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- Cleanup some of the refuel code
  - Removed unnecessary macro
  - Removed duplicate code
  - More comments
  - Fixed a missing private
- Add refueling of containers (so you can refuel the fuel cargo of a fuel truck or a jerry can)

### CODE REVIEW

File: addons/refuel/CfgVehicles.hpp [hunk 1]
```diff
@@ -1,44 +1,5 @@
 #define MACRO_NOZZLE_ACTIONS \
```
Comment (jonpas) [inline addons/refuel/CfgVehicles.hpp]: Should this not also be removed? I don't think anyone could have included this file from a built mod as it has config in it, it had to be copied, so it's safe to remove.

File: addons/refuel/functions/fnc_canTurnOn.sqf [hunk 2]
```diff
@@ -6,6 +6,7 @@
  * Arguments:
  * 0: Unit <OBJECT>
  * 1: Nozzle <OBJECT>
+ * 2: Refuel container (optional) <BOOL>
```
Comment (jonpas) [inline addons/refuel/functions/fnc_canTurnOn.sqf]:
```suggestion
 * 2: Refuel container <BOOL> (default: false)
```

File: addons/refuel/functions/fnc_canTurnOn.sqf [hunk 3]
```diff
@@ -16,15 +17,34 @@
...
+};
+
 !(_nozzle getVariable [QGVAR(isRefueling), false]) &&
-    {[_nozzle getVariable QGVAR(source)] call FUNC(getFuel) != 0} &&
-    {!isNull (_nozzle getVariable [QGVAR(sink), objNull])} &&
-    {(fuel (_nozzle getVariable QGVAR(sink))) < 1}
+    {[_source] call FUNC(getFuel) != 0} &&
+    {!isNull _sink} &&
+    {!_isFull} &&
+    {!(_refuelContainer && {_source == _sink})} && // No endless container ot itself loop
```
Comment (jonpas) [inline addons/refuel/functions/fnc_canTurnOn.sqf]:
```suggestion
    {!(_refuelContainer && {_source == _sink})} && // No endless container of itself loop
```

File: addons/refuel/functions/fnc_disconnect.sqf [hunk 4]
```diff
@@ -1,10 +1,10 @@
 #include "script_component.hpp"
 /*
  * Author: GitHawk
- * Disconnect a fuel nozzle.
+ * Disconnect a fuel nozzle and make unit pick it up.
  *
  * Arguments:
- * 0: Unit <OBJECT>
+ * 0: Unit <OBJECT> (optional)
```
Comment (jonpas) [inline addons/refuel/functions/fnc_disconnect.sqf]:
```suggestion
 * 0: Unit <OBJECT> (default: objNull - drop on floor)
```

File: addons/refuel/functions/fnc_refuel.sqf [hunk 5]
```diff
@@ -20,85 +20,89 @@
...
-private _config = configFile >> "CfgVehicles" >> typeOf _sink;
-
-private _rate =  getNumber (_config >> QGVAR(flowRate)) * GVAR(rate);
-private _maxFuel = getNumber (_config >> QGVAR(fuelCapacity));
-
-// Fall back to vanilla fuelCapacity value (only air and sea vehicles don't have this defined by default by us)
-// Air and sea vehicles have that value properly defined in liters, unlike ground vehicles which is is formula of (range * tested factor) - different fuel consumption system than ground vehicles
-if (_maxFuel == 0) then {
-    _maxFuel = getNumber (_config >> "fuelCapacity");
+private _rate = if (isNumber (configFile >> "CfgVehicles" >> typeOf _sink >> QGVAR(flowRate))) then {
```
Comment (jonpas) [inline addons/refuel/functions/fnc_refuel.sqf]: I hate this if assignment construct, I'd write it as:
```sqf
private _cfgFlowRate = configFile >> "CfgVehicles" >> typeOf _sink >> QGVAR(flowRate);
private _rate = GVAR(rate); // Default to 1 (jerry cans for example have no flow rate defined)
if (isNumber _cfgFlowRate) then {
    _rate = _rate * getNumber _cfgFlowRate;
};
```
Also less commands due to config stored in variable.

But up to you.

**Comment by BrettMayson** [comment]
Completed in #8981, if I am wrong feel free to re-open

**Comment by AutarkOS** [comment]
> Completed in #8981, if I am wrong feel free to re-open

Hi, I'm struggling to work out how this is done in game. I can't attach a nozzle to any fuel container from another. I'm not sure if this just a code capability because I can see that from a console you can set the fuel containers fullness with "ace_refuel_fnc_setFuel"?

**Pull request closed without merging**